### PR TITLE
feat: track details container

### PR DESCRIPTION
**Pull request by sagar-wednesday**
### Ticket Link

---

### Related Links

---

### Description

---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: app/components/TracksCard/index.js [hunk 1]
```diff
@@ -190,8 +202,9 @@ export function TracksCard({ collectionName, trackName, imgUrl, previewUrl, hand
         <CardDescription>{collectionName}</CardDescription>
       </If>
       <ButtonContainer>
-        {/* <MoreButton onClick={(e) => handleRedirectingToPage(e, trackId)}>More</MoreButton> */}
-        <MoreButton>More</MoreButton>
+        <If condition={moreButton}>
```
Comment (praveenkumar1798) [inline app/components/TracksCard/index.js]: isMoreButtonActive will be better

File: app/components/TracksCard/index.js [hunk 2]
```diff
@@ -173,13 +185,13 @@ export function TracksCard({ collectionName, trackName, imgUrl, previewUrl, hand
     }
   };
 
-  // const handleRedirectingToPage = (e, trackId) => {
-  //   e.preventDefault();
-  //   history.push(`/track/${trackId}`);
-  // };
+  const handleRedirectingToPage = (e, trackId) => {
```
Comment (praveenkumar1798) [inline app/components/TracksCard/index.js]: rename to handleRedirectingToTrackDetail

File: app/components/TracksCard/index.js [hunk 3]
```diff
@@ -210,7 +223,10 @@ TracksCard.propTypes = {
   previewUrl: PropTypes.string,
   handleGlobalClick: PropTypes.func,
   tags: PropTypes.string,
-  trackId: PropTypes.number
+  trackId: PropTypes.number,
+  moreButton: PropTypes.bool,
```
Comment (praveenkumar1798) [inline app/components/TracksCard/index.js]: isMoreButtonActive

File: app/containers/SearchTrackContainer/TrackDetailsContainer/index.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+const CustomDetailsContainer = styled.div`
+  background-color: ${colors.secondary};
+  border-radius: 0.5rem;
+  width: 32rem;
+  display: flex;
+  gap: 2rem;
+`;
+
+const CustomExtraDetailsContainer = styled.div`
```
Comment (praveenkumar1798) [inline app/containers/SearchTrackContainer/TrackDetailsContainer/index.js]: CustomTrackDetailContainer will be better

File: app/containers/SearchTrackContainer/TrackDetailsContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  gap: 1rem;
+`;
+
+const CustomExtraDetails = styled.p`
```
Comment (praveenkumar1798) [inline app/containers/SearchTrackContainer/TrackDetailsContainer/index.js]: customTrackDetails

File: app/containers/SearchTrackContainer/index.js [hunk 6]
```diff
@@ -143,6 +140,8 @@ export function SearchTrackContainer({
                 previewUrl={item.previewUrl}
                 tags={item.kind}
                 trackId={item.trackId}
+                moreButton
```
Comment (praveenkumar1798) [inline app/containers/SearchTrackContainer/index.js]: isMoreButtonPresent

File: app/containers/SearchTrackContainer/saga.js [hunk 7]
```diff
@@ -15,7 +17,29 @@ export function* getItunesTracks(action) {
   }
 }
 
+export function* getItunesTrackDetails(action) {
+  let tracksData;
+
+  tracksData = yield select(selectTracksData());
```
Comment (praveenkumar1798) [inline app/containers/SearchTrackContainer/saga.js]: const tracksData = yield select(selectTracksData())

File: app/components/TracksCard/index.js (hunk 3 above)
Comment (sagar-wednesday) [reply app/components/TracksCard/index.js]: okay

File: app/containers/SearchTrackContainer/TrackDetailsContainer/index.js (hunk 4 above)
Comment (sagar-wednesday) [reply app/containers/SearchTrackContainer/TrackDetailsContainer/index.js]: this container is for displaying the extra details than that of the card that's why I did that
 but if you think that is better naming convention than I will change

File: app/containers/SearchTrackContainer/index.js (hunk 6 above)
Comment (sagar-wednesday) [reply app/containers/SearchTrackContainer/index.js]: okay

**Review by praveenkumar1798 — APPROVED** [review]

**Pull request merged**